### ISSUE TITLE
[Q] How to use variables in nested config for custom expression plot?  

### ISSUE THREAD

**Issue by SeminKim** [wandb/wandb#7957]
Hi, I have a question about drawing line plots with expressions.
I am using pytorch lightning and logging hyperparams in a nested dictionary, as shown below (I put the raw file in run config tab here).
I have no problem with using values in the top-level dict(like `${config:seed_everything}`)
Now I want to use the variable `batch_size` for plot. I already tried things like `${config:batch_size}`, `${config:data:batch_size}` or `${config:data/batch_size}` but none of them worked.
What is the correct way to use such variable to plot custom metric?

```
{
  "data": {
    "desc": null,
    "value": {
      "dataset": "imagenet",
      "task_type": "classification",
      "batch_size": 128
    }
  },
  "seed_everything": {
    "desc": null,
    "value": 123
  },
  (...)
}
```

**Comment by vitoria-wandb** [comment]
Hey @SeminKim - could you describe what is your ultimate ideal use case/ workflow? Which kind of metric or custom plot are you trying to log / achieve? Thanks!

**Comment by SeminKim** [comment]
Hi @vitoria-wandb,
I logged the sum of loss in a batch for each training step and now I want to compare the mean loss (mean over single batch) between different runs.
To do this, I want to draw a line plot panel showing mean_loss = (train_loss / batch_size).
As the runs have different batch sizes, I cannot simply scale the train loss with same number.
(I am also facing similar issue, i.e. when I plot the unweighted loss = loss / weight_coefficient).
I tried to draw it in the Expressions tab of line plot, but have no idea about how can I retrieve batch_size from my logged config. 
It seems in wandb it is logged like `{'data':{'value':{'batch_size':128}}}`. I attached the part of raw config in the first comment (result of 'view raw data' button in config tab of run overview).
For the workflow: I have a `Data` class with `batch_size` as an init argument and I am logging all the config with `log_hyperparams` 
function of Pytorch Lightning. Hyperparams and class initialization arguments are automatically handled by LightingCLI, and it will internally log them by calling `wandb.config.update`.
Thanks for your help!

**Comment by vitoria-wandb** [comment]
Hey @SeminKim - I think I see what you mean, let me know if this code can be of help [here](https://colab.research.google.com/drive/1MuwJ5TvpRSdc848T_bIRerDoFMB3vwM7#scrollTo=4zjw0H-oLikY).




Workspace [here](https://wandb.ai/wandb/pytorch-lightning-mnist-2?nw=nwuservitorialima). 

The entire code is in the Colab link above, but have a look at these lines specifically:

```
        self.log('train_loss', loss)
        self.log('mean_loss', mean_loss)
        self.log('batch_size', self.config['data']['value']['batch_size'])
```


I believe you can log the `mean_loss` already in the code, or you can do so from the UI. Please see the video below.

Let me know if this was of help! And let me know if you have any questions!
 

https://github.com/user-attachments/assets/30a3810d-7a6b-4604-a0e6-289159ed9188

**Comment by SeminKim** [comment]
Thanks for your kind reply, it seems I can mitigate the issue by updating my training code to explicitly log `mean_loss` or `batch_size`.
Still, as I already have lots of old runs without explicit logging, it would be great if I can reuse them.
Is there a way to directly use the config's `batch_size` on UI, for  those old experiments? (I guess there's a way to do this because I can utilize this value for filtering runs as shown below...)

![image](https://github.com/user-attachments/assets/daa7b073-eb9e-4b3f-b8ab-ec2fa5caec8e)

**Comment by vitoria-wandb** [comment]
Hi Kim, since we have not heard back from you we are going to close this request. If you would like to re-open the conversation, please let us know!
